Xamarin.Forms.Core/DataTemplateSelector.cs

```diff
@@ -5,18 +5,20 @@ namespace Xamarin.Forms
 {
 	public abstract class DataTemplateSelector : DataTemplate
 	{
-		Dictionary<Type, DataTemplate> _dataTemplates = new Dictionary<Type, DataTemplate>();
+		readonly Dictionary<Type, DataTemplate> _dataTemplates = new Dictionary<Type, DataTemplate>();
+
+		// NOTE: we don't call base as this is a proxy object that does not need to be bound
+		//       and we do not actually make use of any of the base functionality
+		internal override object OnCreateContent(object item, BindableObject container) =>
+			SelectTemplate(item, container)?.CreateContent(item, container);
```

Comment:
Pass the item and container to both the template selector and the template. If different data is needed for some reason, the caller can still use the `SelectTemplate` method as always.